smoke-tests/apps/HttpPreaggregatedMetrics/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpPreaggregatedMetricsSmokeTest.java

```diff
@@ -0,0 +1,237 @@
+
+  private static void verifyHttpClientPreAggregatedMetrics(List<Envelope> metrics)
+      throws Exception {
+    assertThat(metrics.size()).isEqualTo(3);
+    // sort metrics based on result code
+    metrics.sort(
+        Comparator.comparing(
+            obj -> {
+              MetricData metricData = (MetricData) ((Data<?>) obj.getData()).getBaseData();
+              return metricData.getProperties().get("dependency/resultCode");
+            }));
+
+    // 1st pre-aggregated metric
+    Envelope envelope1 = metrics.get(0);
+    validateTags(envelope1);
+    MetricData md1 = (MetricData) ((Data<?>) envelope1.getData()).getBaseData();
+    validateMetricData("client", md1, "200");
+
+    // 2nd pre-aggregated metric
+    Envelope envelope2 = metrics.get(1);
+    validateTags(envelope2);
+    MetricData md2 = (MetricData) ((Data<?>) envelope2.getData()).getBaseData();
+    validateMetricData("client", md2, "404");
+
+    // 3rd pre-aggregated metric
+    Envelope envelope3 = metrics.get(2);
+    validateTags(envelope3);
+    MetricData md3 = (MetricData) ((Data<?>) envelope3.getData()).getBaseData();
+    validateMetricData("client", md3, "500");
+  }
+
+  private static void verifyHttpServerPreAggregatedMetrics(List<Envelope> metrics)
+      throws Exception {
+    assertThat(metrics.size()).isEqualTo(2);
+    // 1st pre-aggregated metric
+    Envelope envelope1 = metrics.get(0);
+    validateTags(envelope1);
+    MetricData md1 = (MetricData) ((Data<?>) envelope1.getData()).getBaseData();
+    validateMetricData("server", md1, "200");
+
+    // 2nd pre-aggregated metric
+    Envelope envelope2 = metrics.get(1);
+    validateTags(envelope2);
+    MetricData md2 = (MetricData) ((Data<?>) envelope2.getData()).getBaseData();
+    validateMetricData("server", md2, "200");
+  }
+
+  private static void validateTags(Envelope envelope) {
+    Map<String, String> tags = envelope.getTags();
+    assertThat(tags.get("ai.internal.sdkVersion")).isNotNull();
+    assertThat(tags).containsEntry("ai.cloud.roleInstance", "testroleinstance");
+    assertThat(tags).containsEntry("ai.cloud.role", "testrolename");
+  }
+
+  private static void validateMetricData(String type, MetricData metricData, String resultCode) {
+    List<DataPoint> dataPoints = metricData.getMetrics();
+    assertThat(dataPoints).hasSize(1);
+    DataPoint dataPoint = dataPoints.get(0);
+    assertThat(dataPoint.getCount()).isEqualTo(1);
+    assertThat(dataPoint.getValue()).isGreaterThan(0d).isLessThan(5 * 60 * 1000d); // (0 - 5) min
+    assertThat(dataPoint.getMin()).isGreaterThan(0d).isLessThan(5 * 60 * 1000d); // (0 - 5) min
+    assertThat(dataPoint.getMax()).isGreaterThan(0d).isLessThan(5 * 60 * 1000d); // (0 - 5) min
+    Map<String, String> properties = metricData.getProperties();
+    String expectedSuccess = "200".equals(resultCode) ? "True" : "False";
+    if ("client".equals(type)) {
+      assertThat(properties.get("_MS.MetricId")).isEqualTo("dependencies/duration");
```

Comment:
can we assert properties size to make sure no new ones slip in without us noticing?